infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/VcsSslCertificateProvisioner.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.kubernetes.provision;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static com.google.common.base.Strings.nullToEmpty;
+import static java.util.Collections.singletonMap;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.inject.Inject;
+import io.fabric8.kubernetes.api.model.ConfigMapBuilder;
+import io.fabric8.kubernetes.api.model.ConfigMapVolumeSourceBuilder;
+import io.fabric8.kubernetes.api.model.Container;
+import io.fabric8.kubernetes.api.model.Volume;
+import io.fabric8.kubernetes.api.model.VolumeBuilder;
+import io.fabric8.kubernetes.api.model.VolumeMount;
+import io.fabric8.kubernetes.api.model.VolumeMountBuilder;
+import java.util.Optional;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import org.eclipse.che.api.core.model.workspace.runtime.RuntimeIdentity;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.eclipse.che.workspace.infrastructure.kubernetes.environment.KubernetesEnvironment;
+import org.eclipse.che.workspace.infrastructure.kubernetes.environment.KubernetesEnvironment.PodData;
+
+/**
+ * Mount configured self-signed certificate for git provider as file in each workspace machines if
+ * configured.
+ *
+ * @author Vitalii Parfonov
+ */
+@Singleton
+public class VcsSslCertificateProvisioner
+    implements ConfigurationProvisioner<KubernetesEnvironment> {
+  static final String CHE_GIT_SELF_SIGNED_CERT_CONFIG_MAP_SUFFIX = "-che-git-self-signed-cert";
+  static final String CHE_GIT_SELF_SIGNED_VOLUME = "che-git-self-signed-cert";
+  static final String CERT_MOUNT_PATH = "/etc/che/git/cert/";
+  static final String CA_CERT_FILE = "cert.pem";
+
+  private static final String HTTPS = "https://";
+
+  @Inject(optional = true)
+  @Named("che.git.self_signed_cert")
+  private String certificate;
+
+  @Inject(optional = true)
+  @Named("che.git.self_signed_cert_host")
+  private String host;
+
+  public VcsSslCertificateProvisioner() {}
+
+  @VisibleForTesting
+  VcsSslCertificateProvisioner(String certificate, String host) {
+    this.certificate = certificate;
+    this.host = host;
+  }
+
+  /** @return true only if */
+  public boolean isConfigured() {
+    return !isNullOrEmpty(certificate);
+  }
+
+  public String getCertPath() {
+    return CERT_MOUNT_PATH + CA_CERT_FILE;
+  }
+
+  public String getGitServerHost() {
+    if (isNullOrEmpty(host)) {
+      return nullToEmpty(host);
```

Comment:
Can you elaborate more about what will happen if host is empty? Is it a valid case? Then git will use such a certificate for everything?